Framework/Algorithms/src/CompareWorkspaces.cpp

```diff
@@ -1066,13 +1067,18 @@ void CompareWorkspaces::doPeaksComparison(PeaksWorkspace_sptr tws1, PeaksWorkspa
       } else {
         g_log.information() << "Column " << name << " is not compared\n";
       }
-      if (std::fabs(s1 - s2) > tolerance) {
+      bool mismatch = false;
+      if (isRelErr && relErr(s1, s2, tolerance)) {
+        mismatch = true;
+      } else if (std::fabs(s1 - s2) > tolerance) {
```

Comment:
If isRelErr=true and the data value passes the relative tolerance check (ie relErr returns false) then the code falls into the else part and performs the absolute check.
The code should only perform one or the other so I think you need to remove the && from the if statement and call relErr instead inside the first part of the if...else statement